code/game/machinery/recharger.dm

```diff
@@ -163,3 +168,90 @@
 			icon_state = "recharger2"
 		return
 	icon_state = "recharger0"
+
+/obj/machinery/recharger/wallrecharger
+	name = "wall recharger"
+	icon = 'icons/obj/wallcharger.dmi'
+	icon_state = "wrecharger0"
+
+/obj/machinery/recharger/wallrecharger/Initialize()
+	. = ..()
+	var/obj/item/weapon/circuitboard/machine/recharger/wallrecharger/B = new()
+	B.apply_default_parts(src)
+
+/obj/item/weapon/circuitboard/machine/recharger/wallrecharger

```

Comment:
This would only be necessary if you're making it constructable.

Reply:
I am making it constructable, and i tested it to be sure it worked and was fully constructable/deconstructable

Reply:
They would both use the same board but you screwdriver the board to switch modes like a few other machines (freezer/heater for example)